### PR TITLE
Include (and print) git version info.

### PR DESCRIPTION
**Pull request by AlCutter**
Addresses #787

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -12,9 +12,16 @@ BUILT_SOURCES = \
...
 	cpp/monitoring/prometheus/metrics.pb.h \
+	cpp/version.h \
 	proto/ct.pb.cc \
 	proto/ct.pb.h
 
+BUILD_VERSION=`git describe --abbrev=0 --dirty --always --tags`
+
+.PHONY: cpp/version.h
+cpp/version.h:
+	echo "const char kBuildVersion[] = \"$(BUILD_VERSION)\";" > $@-t && mv $@-t $@
```
Comment (pphaneuf) [inline Makefile.am]: Put in an `$(AM_V_GEN)`, like for other generated files.

**Comment by pphaneuf** [comment]
Would have preferred a `.cc` file (to avoid recompiling all those somewhat heavier files, but just compile one and relink instead), but seems okay for now.

File: Makefile.am (hunk 1 above)
Comment (AlCutter) [inline Makefile.am]: Done

**Comment by pphaneuf** [comment]
LGTM

**Pull request merged**